src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,136 @@
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+}
+VARIABLES = "temperature_2m_mean,precipitation_sum,soil_moisture_0_to_10cm_mean"
+
+def make_api_request(url: str, params: Dict[str, str])-> Optional[Dict]:
+    """
+    Generic function to make an API request and extract data in JSON format.
+    :param url: API URL
+    :param params: Request parameters
+    :return: Data in JSON format or None if there is an error
+    """
+
+    try:
+        response = requests.get(url, params)
+        response.raise_for_status()
+
+        return response.json()
+    
+    except requests.exceptions.RequestException as e:
+        print(f"Error making API request: {e}")
+        print(f"Response content: {response.text}") 
+        return None
+
+
+def get_data_meteo_api(city: str, start_date: str, end_date: str) -> Optional[Dict]:
+    """
+    Retrieves climate data from an API for a specific city and date range.
+    :param city: City name
+    :param start_date: Start date in "YYYY-MM-DD" format
+    :param end_date: End date in "YYYY-MM-DD" format
+    :return: Climate data in JSON format or None if there is an error
+    """
+    coordinates = COORDINATES.get(city)
+
+    if coordinates is None:
+        print(f"Error: Coordenadas no encontradas para la ciudad {city}")
+        return None
+    
+    latitude, longitude = coordinates['latitude'], coordinates['longitude']
+
+    params ={
+        "latitude": latitude,
+        "longitude": longitude,
+        "start_date": start_date,
+        "end_date": end_date,
+        "models": ["CMCC_CM2_VHR4", "FGOALS_f3_H", "HiRAM_SIT_HR", "MRI_AGCM3_2_S", "EC_Earth3P_HR", "MPI_ESM1_2_XR", "NICAM16_8S"],
+        "daily": VARIABLES
+        }
+
+    data = make_api_request(API_URL, params)
+
+    df = process_climate_data(data)
+
+    return data
+
+
+def process_climate_data(data: Dict) -> pd.DataFrame:
+    """
+    Processes climate data and returns a DataFrame.
+    :param data: Climate data in JSON format
+    :return: Processed DataFrame
+    """
+    daily_data = data.get('daily', {})
+
+    # Crea un DataFrame a partir de los datos diarios
+    df = pd.DataFrame(daily_data)
+
+    # Añade una columna para el tiempo
+    df['time'] = pd.to_datetime(df['time'], format='%Y-%m-%d')
+    df['time'] = df['time'].dt.year
+
+    # Agrupa por año y calcula la media para cada columna
+    df_anual_media = df.groupby(df['time']).mean()
+    df_anual_media.columns = [col + '_media_anual' if col != 'time' else col for col in df_anual_media.columns]
+
+    # Agrupa por año y calcula la std para cada columna
+    df_anual_std = df.groupby(df['time']).std()
+    df_anual_std.columns = [col + '_std_anual' if col != 'time' else col for col in df_anual_std.columns]
+
+    # Hago el dataframe final
+    df = pd.concat((df_anual_media, df_anual_std), axis = 1)
+
+    df.reset_index(inplace=True)
+
+    return df
+
+
+def plot_model_data(parameter: str, models: List[str], df: pd.DataFrame) -> None:
```

Comment:
Everything in one plot. Is it the best way to visualize the data? What can you get out of it? Are we representing comparable things together? Is it clear and help drawing insights or conclusions?